### ISSUE TITLE
Banjo-Tooie: dynarec crashes when switching to the Golden Goliath in level 1

### ISSUE THREAD

**Issue by BonzaiThePenguin** [mupen64plus-ae/mupen64plus-ae#410]
This one is 100% reproducible and only affects the dynarec. When Mumbo raises the Golden Goliath in Mayahem Temple (the first level), you're supposed to take control of the Goliath when the cutscene is finished. Instead the game crashes and returns you to the Resume/Restart/Settings menu.

Workaround is to freeze state the game immediately before the cutscene ends, switch to either interpreter for the switch to the Golden Goliath, then switch back to the dynarec afterwards.

**Comment by littleguy77** [comment]
Yes, I believe this is one of the fixes @Gillou68310 is working on but not sure if he likes the solution:
https://github.com/Gillou68310/mupen64plus-ae/issues/1#issuecomment-68690556

**Comment by BonzaiThePenguin** [comment]
Ha, for some reason I completely forgot that Banjo-Tooie also randomly crashes every few minutes. Yeah I get that too.

Does this app write out the chunk of code the dynarec had been working on when it crashed, to make it easier to figure out which instruction(s) needs to be fixed? I checked the source and nothing immediately jumped out at me as doing this, but I didn't check very thoroughly.

**Comment by BonzaiThePenguin** [comment]
Also, I took a freeze state of a section of the game that also crashes 100% of the time for me (IIRC, it was switching to Mumbo in Terrydactyland), but what's strange is that the first time I played the level it worked perfectly. But then the game did one of its random crashes, and when I replayed to get back to Mumbo it started crashing when switching to him. Even stranger, when I load from the freeze state Mumbo's house is invisible and I'm pretty sure the camera angle is different from what it normally is.

But yeah, if there's a way to narrow down the section of code where the dynarec starts to differ from the interpreter, I wouldn't mind taking a crack at it.
